<br>

<br>

# 👩🏻‍👩🏻‍👧🏻‍👦🏻 **SOCIODEMOGRAPHIC AND HEALTH RESOURCE DATA**👩🏻‍👩🏻‍👧🏻‍👦🏻

**LINEAR REGRESSION MODEL USING PANDAS AND PYTHON**

<br>

## **INDEX**

- **STEP 1: PROBLEM DEFINITION AND DATA COLLECTION**
- **STEP 2: DATA EXPLORATION AND CLEANING**
- **STEP 3: UNIVARIATE VARIABLE ANALYSIS**


<br>

## **STEP 1: PROBLEM DEFINITION AND DATA COLLECTION**

- 1.1. Problem Definition
- 1.2. Library Importing
- 1.3. Data Collection

**1.1. PROBLEM DEFINITION**

This project analyzes US county-level sociodemographic and health resource data collected between 2018 and 2019.  Understanding the interplay between **these factors is crucial for informing public health policy, resource allocation, and intervention strategies.**

Variations in health outcomes across different counties may be influenced by both the availability of health resources and the specific sociodemographic characteristics of the population.



**Problem Statement:**  The core problem this project addresses is determining the relationship between sociodemographic factors and health resources at the county level in the United States.  Specifically, we aim to identify which sociodemographic variables are most strongly associated with the availability and utilization of health resources.

<br>

**1.2. LIBRARY IMPORTING**

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.feature_selection import SelectKBest, f_regression
from sklearn.linear_model import LinearRegression, Lasso
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error
import warnings
warnings.filterwarnings("ignore")

**1.3. DATA COLLECTION**

In [6]:
df = pd.read_csv("https://raw.githubusercontent.com/4GeeksAcademy/regularized-linear-regression-project-tutorial/main/demographic_health_data.csv")
df.head()

fips  TOT_POP    0-9  0-9 y/o % of total pop  19-Oct  \
0  1001    55601   6787               12.206615    7637   
1  1003   218022  24757               11.355276   26913   
2  1005    24881   2732               10.980266    2960   
3  1007    22400   2456               10.964286    2596   
4  1009    57840   7095               12.266598    7570   

   10-19 y/o % of total pop  20-29  20-29 y/o % of total pop  30-39  \
0                 13.735364   6878                 12.370281   7089   
1                 12.344167  23579                 10.814964  25213   
2                 11.896628   3268                 13.134520   3201   
3                 11.589286   3029                 13.522321   3113   
4                 13.087828   6742                 11.656293   6884   

   30-39 y/o % of total pop  ...  COPD_number  diabetes_prevalence  \
0                 12.749771  ...         3644                 12.9   
1                 11.564429  ...        14692                 12.0   
2                 12.865239  ...         2373                 19.7   
3                 13.897321  ...         1789                 14.1   
4                 11.901798  ...         4661                 13.5   

   diabetes_Lower 95% CI  diabetes_Upper 95% CI  diabetes_number  \
0                   11.9                   13.8             5462   
1                   11.0                   13.1            20520   
2                   18.6                   20.6             3870   
3                   13.2                   14.9             2511   
4                   12.6                   14.5             6017   

   CKD_prevalence  CKD_Lower 95% CI  CKD_Upper 95% CI  CKD_number  \
0             3.1               2.9               3.3        1326   
1             3.2               3.0               3.5        5479   
2             4.5               4.2               4.8         887   
3             3.3               3.1               3.6         595   
4             3.4               3.2               3.7        1507   

   Urban_rural_code  
0                 3  
1                 4  
2                 6  
3                 2  
4                 2  

[5 rows x 108 columns]

<br>

## **STEP 2: DATA EXPLORATION AND CLEANING**

- 2.1. Data Exploration.
- 2.2. Data Cleaning.
- 


<br>

**2.1. DATA EXPLORATION**

In [22]:
df.shape

(3140, 108)

This DataFrame contains **3140 ROWS** and **108 COLUMNS**

<br>

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3140 entries, 0 to 3139
Columns: 108 entries, fips to Urban_rural_code
dtypes: float64(61), int64(45), object(2)
memory usage: 2.6+ MB


In [17]:
df.describe().transpose()

count           mean            std     min  \
fips                    3140.0   30401.640764   15150.559265  1001.0   
TOT_POP                 3140.0  104189.412420  333583.395432    88.0   
0-9                     3140.0   12740.302866   41807.301846     0.0   
0-9 y/o % of total pop  3140.0      11.871051       2.124081     0.0   
19-Oct                  3140.0   13367.976752   42284.392134     0.0   
...                        ...            ...            ...     ...   
CKD_prevalence          3140.0       3.446242       0.568059     1.8   
CKD_Lower 95% CI        3140.0       3.207516       0.527740     1.7   
CKD_Upper 95% CI        3140.0       3.710478       0.613069     1.9   
CKD_number              3140.0    2466.234076    7730.422067     3.0   
Urban_rural_code        3140.0       4.635350       1.510447     1.0   

                                 25%           50%          75%           max  
fips                    18180.500000  29178.000000  45081.50000  5.604500e+04  
TOT_POP                 10963.250000  25800.500000  67913.00000  1.010552e+07  
0-9                      1280.500000   3057.000000   8097.00000  1.208253e+06  
0-9 y/o % of total pop     10.594639     11.802727     12.95184  2.546068e+01  
19-Oct                   1374.500000   3274.000000   8822.25000  1.239139e+06  
...                              ...           ...          ...           ...  
CKD_prevalence              3.100000      3.400000      3.80000  6.200000e+00  
CKD_Lower 95% CI            2.900000      3.200000      3.50000  5.800000e+00  
CKD_Upper 95% CI            3.300000      3.700000      4.10000  6.600000e+00  
CKD_number                314.750000    718.000000   1776.25000  2.377660e+05  
Urban_rural_code            3.000000      5.000000      6.00000  6.000000e+00  

[106 rows x 8 columns]

In [20]:
df.nunique()

fips                      3140
TOT_POP                   3074
0-9                       2723
0-9 y/o % of total pop    3136
19-Oct                    2743
                          ... 
CKD_prevalence              43
CKD_Lower 95% CI            39
CKD_Upper 95% CI            46
CKD_number                1894
Urban_rural_code             6
Length: 108, dtype: int64

In [27]:
print("THESE ARE ALL THE COLUMNS OF THE DATASET:")
print(df.columns.tolist())

THESE ARE ALL THE COLUMNS OF THE DATASET:
['fips', 'TOT_POP', '0-9', '0-9 y/o % of total pop', '19-Oct', '10-19 y/o % of total pop', '20-29', '20-29 y/o % of total pop', '30-39', '30-39 y/o % of total pop', '40-49', '40-49 y/o % of total pop', '50-59', '50-59 y/o % of total pop', '60-69', '60-69 y/o % of total pop', '70-79', '70-79 y/o % of total pop', '80+', '80+ y/o % of total pop', 'White-alone pop', '% White-alone', 'Black-alone pop', '% Black-alone', 'Native American/American Indian-alone pop', '% NA/AI-alone', 'Asian-alone pop', '% Asian-alone', 'Hawaiian/Pacific Islander-alone pop', '% Hawaiian/PI-alone', 'Two or more races pop', '% Two or more races', 'POP_ESTIMATE_2018', 'N_POP_CHG_2018', 'GQ_ESTIMATES_2018', 'R_birth_2018', 'R_death_2018', 'R_NATURAL_INC_2018', 'R_INTERNATIONAL_MIG_2018', 'R_DOMESTIC_MIG_2018', 'R_NET_MIG_2018', 'Less than a high school diploma 2014-18', 'High school diploma only 2014-18', "Some college or associate's degree 2014-18", "Bachelor's degree or hi

<br>

**2.2. DATA CLEANING**

In [31]:
# Check for missing values
df.isnull().sum()

fips                      0
TOT_POP                   0
0-9                       0
0-9 y/o % of total pop    0
19-Oct                    0
                         ..
CKD_prevalence            0
CKD_Lower 95% CI          0
CKD_Upper 95% CI          0
CKD_number                0
Urban_rural_code          0
Length: 108, dtype: int64

**OBSERVATIONS:** The result shows that no columns in the DataFrame have any missing values.

<br>

In [32]:
# Drop irrelevant columns
df.drop(columns=['fips'], inplace=True)

print("\nUpdated Dataset Shape after dropping irrelevant columns:", df.shape)



Updated Dataset Shape after dropping irrelevant columns: (3140, 107)


- **Why 'fips'?** Because it's just a county identifier and doesn't help with the predictive analysis.
- **Effect:** The dataset shape changes from (3140, 108) to (3140, 107), confirming that one column was removed.

<br>

In [36]:
duplicates = df.duplicated().sum()
print(f"\nNUMBER OF DUPLICATE ROWS: {duplicates}")


NUMBER OF DUPLICATE ROWS: 0


In [37]:
# Convert column names to lowercase and replace spaces with underscores
df.columns = df.columns.str.lower().str.replace(' ', '_')


**Data Formatting**
- Ensure all categorical variables are properly encoded.
- Standardize column names (if needed).

<br>

## **STEP 3: UNIVARIATE VARIABLE ANALYSIS**

- 3.1.

**OBJECTIVE OF UNIVARIATE VARIABLE ANALYSIS:** Univariate variable analysis focuses on examining each variable individually to understand its distribution, central tendency, spread, and potential anomalies. This step helps in:

-  Identifying data patterns (normal vs. skewed distributions).
-  Detecting outliers that may affect model performance.
-  Understanding variable importance before feature selection.
-  Deciding on transformations (e.g., normalization, log scaling) for better model performance.
